pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -36,14 +36,14 @@ const (
 
 // Config for query_range middleware chain.
 type Config struct {
-	SplitQueriesByInterval time.Duration `yaml:"split_queries_by_interval" category:"advanced"`
-	AlignQueriesWithStep   bool          `yaml:"align_queries_with_step"`
-	ResultsCacheConfig     `yaml:"results_cache"`
-	CacheResults           bool   `yaml:"cache_results"`
-	MaxRetries             int    `yaml:"max_retries" category:"advanced"`
-	ShardedQueries         bool   `yaml:"parallelize_shardable_queries"`
-	CacheUnalignedRequests bool   `yaml:"cache_unaligned_requests" category:"advanced"`
-	TargetSeriesPerShard   uint64 `yaml:"query_sharding_target_series_per_shard"`
+	SplitQueriesByInterval           time.Duration `yaml:"split_queries_by_interval" category:"advanced"`
+	AlignQueriesWithStep             bool          `yaml:"align_queries_with_step"`
+	ResultsCacheConfig               `yaml:"results_cache"`
+	CacheResults                     bool   `yaml:"cache_results"`
+	MaxRetries                       int    `yaml:"max_retries" category:"advanced"`
+	ShardedQueries                   bool   `yaml:"parallelize_shardable_queries"`
+	DeprecatedCacheUnalignedRequests bool   `yaml:"cache_unaligned_requests" category:"advanced" doc:"hidden"` // TODO: Deprecated in Mimir 2.10.0, remove in Mimir 2.12.0
```

Comment:
When someone sets this field in their config, we should modify the default limit value.

We should do that in `initRuntimeConfig`, before calling `validation.SetDefaultLimitsForYAMLUnmarshalling`, similar to https://github.com/grafana/mimir/blob/52d9875173e1c35f5a4ca5d8c19a9faf87023744/pkg/mimir/modules.go#L273-L279

Reply:
this is addressed now;

Is there a good place in the docs to put a little playbook for migrating a config option into the `limits` section?

There are quite a few checklist items for this process, and I would expect this is far from the last time it will be done.

Reply:
> Is there a good place in the docs to put a little playbook for migrating a config option into the `limits` section?

That's a great idea. We can create short document in `docs/internal/contributing` directory.